examples/metalk8s-solution-example/operator/pkg/controller/util/baseserver.go

```diff
@@ -108,16 +109,28 @@ func BuildLabelSelector(labels map[string]string) metav1.LabelSelector {
 
 // buildImageName builds a complete image name based on the version provided
 // for the `base-server` component, which is the only one deployed for now
+// Here version is both the image and the Solution versions
 func buildImageName(version string) string {
-	prefix, found := os.LookupEnv("REGISTRY_PREFIX")
-	if !found {
-		prefix = "docker.io/metalk8s"
+	var imageName string = "base-server:" + version
+	var prefix string
+
+	for solution_version, repositories := range config.Repositories() {
+		if solution_version == version {
+			for _, repository := range repositories {
+				for _, image := range repository.Images {
+					if image == imageName {
+						prefix = repository.Endpoint
+					}
+				}
+			}
+		}
+	}
+
+	if prefix == "" {
+		return imageName
 	}
```

Comment:
This will happen if an image is not registered in the list of `Repositories` for the given Solution version. I'd rather fail in the Operator, rather than rely on the fact that _it could work, maybe_.

Reply:
Yes, wasn't sure how to handle this.
Will change it and make it fail.

Reply:
done